docs/v6-migration.md

```diff
@@ -0,0 +1,103 @@
+:::
+
+### Zod (Config.js Validation)
+
+The first and most dramatic change many of you will notice, potentially on the first launch after updating, is that you will now receive very detailed error messages when your config file, permissions, paths, and formatting/syntax are incorrect. This (Zod) is a feature we introduced with v6 to make sure you were given more detailed information about why your config file is wrong, without necessarily requiring additional/outside help.
+
+The error messages will give you the general issue as well as link you to the documentation page for the setting that is set incorrectly, give you specific errors on bad paths or permissions, and if you've made a mistake with syntax somewhere - it will also tell you where to look.
+
+This is not going to automatically fix anything for you, but will give you a better starting point to try and solve the issue yourself before requiring outside assistance.
+
+### Linking
+
+We have made drastic breaking changes to the way linking operates, both in its implementation and in expanding its capabilities. Not only is linking more versatile in what can be matched now (for instance, previously files inside a folder would not match to a single-file torrent - this has been fixed), but you can also take advantage of linking when searching .torrent files instead of solely applying to data-based matching.
+
+To achieve torrent linking, you simply have to mount or give permission (if necessary) for `cross-seed` to read and write to the actual torrent data structure (downloaded files) your client uses. This will need to mirror (for docker) the mounts made in the client. Your client will also need to be able to read and write to the `linkDir` folder. You do not need to specify a `dataDirs` with this setup, but simply define a `linkDir`.
+
+Another change that is made is with the linking folder structure found inside of the `linkDir`. By default, everything linked will be made inside of a folder with the name of the indexer it matched to inside of the `linkDir`. (Example: /linkdir/Indexer 1/Torrent.Name.mkv) This will not cause any issues for previously cross-seeded torrents in your client, and will not need changes made to the mounts or permissions you already have set.
+
+There is, however, a new setting introduced into `config.js` named `legacyLinking`. This is an option introduced mainly to accommodate users of qBittorrent who want to utilize the Auto Torrent Management feature. This will link in the flat-folder style previously seen in v5. If you want the old (v5) style behavior of linking or use AutoTMM, then you will need to set this option to `true`.
+
+### torrentDir
+
+Previously, our recommendation if you wanted to strictly search only `dataDirs` for matches was to point `torrentDir` at a empty folder. This is no longer necessary. You can now set `torrentDir` to `null` to achieve a data-only search.

```

Comment:
```suggestion
There is a new way to set nullify the `torrentDir` if you want to run a data-only search using the paths set in `dataDirs`. Set the `torrentDir` setting to `null` (`torrentDir: null`).
```

Reply:
Not sure i entirely agree with using ""nullify" the torrentDir" - as some people might not understand what that accomplishes or why you would want to do this.

Reply:
Sure, that's fair.